pyfar/dsp/regularization.py

```diff
@@ -5,333 +5,373 @@
+            >>> regularization = Inversion.regularization * Inversion.beta
+            ...
+            >>> pf.plot.use()
+            >>> fig, axes = plt.subplots(2,1)
+            >>> pf.plot.freq(sweep, ax=axes[0], label='sweep')
+            >>> pf.plot.freq(regularization, ax=axes[0], label='regularization')
+            >>> axes[0].axvline(50, color='k', linestyle='--',
+            ...     label='frequency range')
+            >>> axes[0].axvline(16e3, color='k', linestyle='--')
+            >>> axes[0].legend(loc='lower center')
+            ...
+            >>> pf.plot.freq(inverted, ax=axes[1], color='p',
+            ...     label='inverted with regulariztion')
+            >>> pf.plot.freq(1 / (sweep+1e-10), ax=axes[1], color='y',
+            ...     linestyle=':', label='inverted without regulariztion')
+            >>> axes[1].axvline(50, color='k', linestyle='--',
+            ...     label='frequency range')
+            >>> axes[1].axvline(16e3, color='k', linestyle='--')
+            >>> axes[1].set_ylim(-120, -20)
+            >>> axes[1].legend(loc='lower center')
+        """# noqa: E501
+        instance = cls.__new__(cls)
+
+        instance._frequency_range = np.asarray(frequency_range).flatten()
+
+        if instance._frequency_range.shape != (2, ):
             raise ValueError(
                 "The frequency range needs to specify lower and upper limits.")
 
-        instance._frequency_range = np.asarray(frequency_range)
+        instance._regularization_type = "frequency range"
+        instance._regularization_within = regularization_within
+        instance.signal = signal
+        instance.beta = beta
+        instance.target = target
 
         return instance
 
     @classmethod
-    def from_magnitude_spectrum(cls, regularization):
-        """
-        Compute regularization from magnitude spectrum.
+    def from_magnitude_spectrum(cls, signal, regularization, beta=1,
+                                target=None):
+        r"""
+        Regularization from a given magnitude spectrum.
 
         Regularization passed as :py:class:`~pyfar.Signal`. The length of
-        `regularization` must match the length the signal to be inverted.
-
-        Parameters
-        ----------
-        regularization : Signal
-            Signal containing the regularization factors. The signal's
-            magnitude spectrum is used for regularization.
-        """
-        if not isinstance(regularization, pf.Signal):
-            raise ValueError(
-                "Regularization must be a pyfar.Signal object.")
-
-        instance = cls(regu_type = "magnitude spectrum")
-        instance._regu = regularization
-
-        return instance
-
-    def invert(self, signal, beta=1, target=None,
-               normalize_regularization=None):
-        r"""
-        Invert the input using frequency-dependent regularized inversion.
+        `regularization` must match the length of the signal to be inverted.
```

Comment:
Is the length the only property that needs to match? Probably also sampling rate?

Reply:
Ah yes, good point.